assets/static/filters.js

```diff
@@ -64,12 +68,65 @@ function setPause() {
     $(selectors.icon).removeClass("fa-circle-o-notch fa-spin fa-search").addClass("fa-pause");
 }
 
+function renderHistory() {
+    const storage = window.localStorage;
+
+    var historicFilters = [];
+
+    const currentFilterText = getFilters().join(",");
+
+    const history = storage.getItem(historyKey);
+    if (history) {
+        historicFilters = history.split("\n");
+    }
+
+    var historyMenuHTML = templates.renderTemplate("historyMenu", {
+        activeFilter: currentFilterText,
+        defaultFilter: $(selectors.filter).data("default-filter"),
+        savedFilter: Cookies.get("defaultFilter.v2"),
+        filters: historicFilters
+    });
+    $(selectors.historyMenu).html(historyMenuHTML);
+}
+
+function appendFilterToHistory(text) {
+    // require non empty text and enabled appends
+    if (!text || !appendsEnabled) return false;
+
+    const storage = window.localStorage;
+
+    // final filter list we'll save to storage
+    var filterList = [ text ];
+
+    // get current history list from storage and append it to our final list
+    // of filters, but avoid duplicates
+    const history = storage.getItem(historyKey);
+    if (history) {
+        const historyArr = history.split("\n");
+        for (var i = 0; i < historyArr.length; i++) {
+            var h = historyArr[i];
+            if (filterList.indexOf(h) < 0) {
+                filterList.push(h);
+            }
+        }
+    }
+
+    // truncate the history to up to 11 elements
+    filterList = filterList.slice(0, 11);
+
+    storage.setItem(historyKey, filterList.join("\n"));
```

Comment:
You'll have to marshal it back here. You might be interested in wrapping the storage engine to deal with marshalling tasks.